src/metric/CompoundMetric.py

```diff
@@ -0,0 +1,129 @@
+            cmm (CompoundMetricManager): Manager used by metrics. Population and dataset is derived
+                from this manager.
+            smm (SampleMetricManager, optional): Manager used by per-sample metrics. Defaults to None.
+        """
+        self._dataset = compound_metric_manager.get_dataset()
+        self._population = compound_metric_manager.get_population()
+        self._name = name
+        self._smm = smm
+        pass
+
+    def get_dataset(self) -> Dataset:
+        """
+        Returns the dedicated dataset of this compound metric. This Dataset
+        was assigned upon the metrics's instantiation, inherited by the compound
+        metric manager.
+
+        Returns:
+            Dataset: The dataset of this manager.
+        """
+        return self._dataset
+
+    def get_population(self) -> Population:
+        """
+        Returns the dedicated population of this compound metric. This Population
+        was assigned upon the metrics's instantiation, inherited by the compound
+        metric manager.
+
+        Returns:
+            Population: The population of this manager.
+        """
+        return self._population
+
+    def get_name(self) -> str:
+        """
+        Should return the name of this compound metric.
+
+        Returns:
+            str: The name of this compound metric.
+        """
+        return self._name
+
+    @abc.abstractmethod
+    def setup(self, mode: str) -> None:
+        """
+        Should train/prepare all auxiliary models required for performing
+        a calculation with `calc()`.
+
+            mode (str): The mode of the setup. The available modes
+                depend on the the metric.
+        """
+        pass
+
+    @abc.abstractmethod
+    def is_ready(self) -> bool:
+        """
+        Should return `True` if the `setup()` has been completed, i.e.,
+        if all auxiliary models are at disposal when calling `calc()`.
+
+        Returns:
+            bool: `True` if the metric is set up.
+        """
+        pass
+
+    @abc.abstractmethod
+    def calc(self, **parameters: Any) -> Any:
+        """
+        Should calculate a compound metric for a set of samples.
+
+        Args:
+            **parameters (Any): Arbitrary metric parameters.
+
+        Returns:
+            Any: Return the value of the metric.
+        """
+        pass
+
+    @abc.abstractmethod
+    def get(self, calc_if_missing: bool = False, **parameters: Any) -> Any:
+        """
+        Get the metric.
+
+        Args:
+            calc_if_missing (bool, optional): If True calculates the metric if missing
+                by calling `calc()`. Defaults to False.
+            **parameters (Any): Arbitrary parameters required by the metrics.
+                These are only used if `calc_if_missing=True`, and are all forwarded
+                to `calc()`.
+
+        Returns:
+            Any: Return the value of the metric. Should return None if the metric is missing
```

Comment:
Redundant "Return...", start with "The..."

Reply:
agree